pkg/api/admin/platformworkloadidentityroleset_convert.go

```diff
@@ -56,12 +60,16 @@ func (c platformWorkloadIdentityRoleSetConverter) ToInternal(_new interface{}, o
 	out.Properties.OpenShiftVersion = new.Properties.OpenShiftVersion
 	out.Properties.PlatformWorkloadIdentityRoles = make([]api.PlatformWorkloadIdentityRole, 0, len(new.Properties.PlatformWorkloadIdentityRoles))
 
-	for i, r := range new.Properties.PlatformWorkloadIdentityRoles {
-		out.Properties.PlatformWorkloadIdentityRoles[i].OperatorName = r.OperatorName
-		out.Properties.PlatformWorkloadIdentityRoles[i].RoleDefinitionName = r.RoleDefinitionName
-		out.Properties.PlatformWorkloadIdentityRoles[i].RoleDefinitionID = r.RoleDefinitionID
-		out.Properties.PlatformWorkloadIdentityRoles[i].ServiceAccounts = make([]string, 0, len(r.ServiceAccounts))
-		out.Properties.PlatformWorkloadIdentityRoles[i].ServiceAccounts = append(out.Properties.PlatformWorkloadIdentityRoles[i].ServiceAccounts, r.ServiceAccounts...)
+	for _, r := range new.Properties.PlatformWorkloadIdentityRoles {
```

Comment:
Why are s.Properties and new.properties basically identical? Is there a way we can implement this once such as using generics?

Reply:
I think in order to leverage generics here `api.PlatformWorkloadIdentityRoleSet`, the internal type, and `admin.PlatformWorkloadIdentityRoleSet`, the admin API type, would have to implement a common interface. If we could do that and wanted to do that, I think that refactor would be outside of the scope of this PR since we'd want to makesimilar improvements across all of the various API converters (like https://github.com/Azure/ARO-RP/blob/master/pkg/api/admin/openshiftcluster_convert.go for example).

Reply:
Thanks for the clarification. We'll put that on the maybe someday list.